### PR TITLE
[cnd] cnd.debugger.common2 and cnd.debugger.gdb2

### PR DESCRIPTION
**Pull request by vieiro**
Recovering cnd.debugger.common2 and cnd.debugger.gdb2 from the contrib area.

- Dropped some dependencies with cnd.model (we won't have auto and macro expansions)
- Tests should be passing, we've dropped the license headers from golden files and added `cnd/cnd.debugger/licenseinfo.xml`
- We'll need to create and add a binary dependency with the executable generated from `cnd/cnd.debugger.common2/windows/` (in a future PR).

We may want to add cnd.debugger.dbx a try later on, for some other specific platforms (Solaris? IBM?)

### CODE REVIEW

**Comment by vieiro** [comment]
D'oh, I forgot the mandatory photo...

cnd debugging

**Comment by mbien** [comment]
I see things are coming together :)

**Comment by vieiro** [comment]
Hi @matthiasblaesing , now that "All checks have passed", would you please take a look a this one?

File: cnd/cnd.debugger.common2/licenseinfo.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+-->
+<licenseinfo>
+    <fileset>
+        <!-- Tests expect this file without an ASF License header -->
+        <file>test/unit/data/goldenfiles/org/netbeans/modules/cnd/debugger/common2/debugger/breakpoints/BreakpointsCodecTest/Breakpoints_new.xml</file>
+        <file>test/unit/data/goldenfiles/org/netbeans/modules/cnd/debugger/common2/debugger/breakpoints/BreakpointsCodecTest/Breakpoints_old.xml</file>
+        <file>test/unit/data/goldenfiles/org/netbeans/modules/cnd/debugger/common2/debugger/breakpoints/BreakpointsCodecTest/Breakpoints_saved.xml</file>
+        <license ref="Apache-2.0-ASF" />
+        <comment type="TEMPLATE_MINIMAL_IP" />
```
Comment (matthiasblaesing) [inline cnd/cnd.debugger.common2/licenseinfo.xml:29]: These are not really templates. I suggest to add a new entry to the `COMMENT_TYPE` enum like this:

```diff
# This patch file was generated by NetBeans IDE
# It uses platform neutral UTF-8 encoding and \n newlines.
--- a/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/licenseinfo/CommentType.java
+++ b/nbbuild/antsrc/org/netbeans/nbbuild/extlibs/licenseinfo/CommentType.java
@@ -22,7 +22,8 @@
     TEMPLATE_MINIMAL_IP("The file is used as a template and contains minimal IP."),
     CATEGORY_B("The contained code is used at runtime and must be included in source form."),
     COMMENT_UNSUPPORTED("Filetype does not support comments."),
-    GUI_USABILITY("Code is visible at runtime and added license would negatively affect usability.");
+    GUI_USABILITY("Code is visible at runtime and added license would negatively affect usability."),
+    TEST_DATA("The file is used in tests and license header can't be added.");
     
     private String outputComment;
     
```

File: cnd/cnd.debugger.common2/src/org/netbeans/modules/cnd/debugger/common2/debugger/Autos.java [hunk 2]
```diff
@@ -22,30 +22,29 @@
 import java.util.Collections;
 import java.util.Set;
 import javax.swing.text.StyledDocument;
-import org.netbeans.modules.cnd.spi.model.services.AutosProvider;
-import org.openide.util.Lookup;
 
 public final class Autos {
```
Comment (matthiasblaesing) [inline cnd/cnd.debugger.common2/src/org/netbeans/modules/cnd/debugger/common2/debugger/Autos.java:26]: Just out of curiosity - what are `Autos`? Some kind of variable?

**Review by matthiasblaesing — APPROVED** [review]
I only looked into this from a project perspective, but the changes look surprisingly minimal. Someone created sane interfaces that allow cuts at a sensible level.

I one inline comment regarding the `licenseinfo.xml`. One thing, that might be nice would be the addition of entries for the graphics files to the `licenseinfo.xml` file. Not necessary, but nice.

**Comment by vieiro** [comment]
Thanks for the review, @matthiasblaesing .
I already have all image files in licenseinfo.xml, but I thought I could add the windows binary tool in the `windows/` directory too, [so I uploaded GdbKillProc.exe to OSUOSL](https://netbeans.osuosl.org/binaries/7315EB1AC148941FD18F9F1F71BD1DBECE67A501-GdbKillProc.exe). 
I'm trying out gdb on Windows 10, but I'm stuck at [something similar to this old 258292 issue](https://bz.apache.org/netbeans/show_bug.cgi?id=258292). 
I'll update the PR once it's solved and we can run gdb on Windows too (gdb in MSYS64 is failing with a 127 error code in my box :-( , and cygwin gdb doesn't work either.

File: cnd/cnd.debugger.common2/src/org/netbeans/modules/cnd/debugger/common2/debugger/Autos.java (hunk 2 above)
Comment (vieiro) [reply cnd/cnd.debugger.common2/src/org/netbeans/modules/cnd/debugger/common2/debugger/Autos.java:26]: `auto`  is a [keyword in C89](https://en.wikipedia.org/wiki/C_(programming_language)) (seldomly used, though).  It was later [enhanced in C++11](https://docs.microsoft.com/en-us/cpp/cpp/auto-cpp?view=msvc-170) and In C++14 they aded the so called [type deduction with braced initializers](https://docs.microsoft.com/en-us/cpp/cpp/auto-cpp?view=msvc-170#type-deduction-with-braced-initializers-c14), as horrible as that looks! :-D

**Comment by vieiro** [comment]
@matthiasblaesing , would you please make a review of the changes?
- Added "TEST_DATA" to the list of comment types for licenseinfo.xml.
- A patch that closes #3947 , making Cygwin gdb's working again. This was due to wrong paths being sent to the debugger.

**Comment by vieiro** [comment]
NetBeans CND sporting the Cygwin's gdb debugger on Windows (NB8.2 failed at this, see #3947 ).
On the screenshot, a project configured with C++14 and a function with an `auto` argument. gdb seems to properly find the value of this variable.
imagen

**Review by matthiasblaesing — APPROVED** [review]
Thank you for the changes. Looks sane to me. 

For GdbKillProc I'm inclined to rewrite it in java using JNA at some later stage. The functions look trivial to bind. This however does not invalidate the work done here and should not stop this from being merged.

**Comment by vieiro** [comment]
+1 to JNA. I added an enhancement here: https://github.com/apache/netbeans/issues/3964

**Pull request merged**